File: frontend/src/services/movieService.js
import { apiClient } from "./apiClient";

export class MovieService {

    getAllDisplayingMovies() {
        return apiClient.get("/movie/movies/displayingMovies");
    }

    getAllComingSoonMovies() {
        return apiClient.get("/movie/movies/comingSoonMovies");
    }

    getAllMovies() {
        return apiClient.get("/movie/movies/all");
    }

    getMoviesByCity(cityName) {
        return apiClient.get("/movie/movies/city/" + encodeURIComponent(cityName));
    }

    getMovieById(movieId) {
        return apiClient.get("/movie/movies/" + movieId);
    }

    addMovie(movieDto) {
        return apiClient.post("/movie/movies/add", movieDto);
    }

    deleteMovie(movieId) {
        return apiClient.delete("/admin/movies/" + movieId);
    }
}

File: movieService/src/main/java/com/doffeii/movieService/business/abstracts/MovieService.java
package com.doffeii.movieService.business.abstracts;

import com.doffeii.movieService.entity.Movie;
import com.doffeii.movieService.entity.dto.MovieRequestDto;
import com.doffeii.movieService.entity.dto.MovieResponseDto;

import java.util.List;

public interface MovieService {

    List<MovieResponseDto> getAllDisplayingMoviesInVision();

    List<MovieResponseDto> getAllComingSoonMovies();

    List<MovieResponseDto> getAllMovies();

    List<MovieResponseDto> getMoviesByCity(String cityName);

    MovieResponseDto getMovieByMovieId(int movieId);

    Movie getMovieById(int movieId);

    Movie addMovie(MovieRequestDto movieRequestDto);

    void deleteMovie(int movieId, String authorizationHeader);
}
